db-client-java/src/main/java/com/eventstore/dbclient/ExpectedRevision.java

```diff
@@ -43,14 +43,17 @@ public static ExpectedRevision expectedRevision(long revision) {
         return new SpecificExpectedRevision(revision);
     }
 
-    static ExpectedRevision fromRaw(long revision) {
+    public static ExpectedRevision fromRawLong(long revision) {
         if (revision == -1)
             return ExpectedRevision.noStream();
         if (revision == -2)
             return ExpectedRevision.any();
         if (revision == -4)
             return ExpectedRevision.streamExists();
 
+        if (revision < 0)
+            throw new RuntimeException(String.format("Invalid expected revision long representation '%s'", revision));
```

Comment:
Side question not 100% related to this PR but how do you actually catch this? I always get another exception for example an unknown exception: 
'Exception in thread "main" java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNKNOWN: Exception was thrown by handler'

using:
 AppendToStreamOptions appendToStreamOptions = AppendToStreamOptions.get()
                .expectedRevision(ExpectedRevision.fromRawLong(-3));
        WriteResult writeResult = eventStore
                .appendToStream("accounts", appendToStreamOptions, event)
                .get();

Reply:
Just in case, there is a test in that PR and you could see that I catch the `RuntimeException` directly. I really don't see what you situation could have been.